File: test/NightlyRun/IdFromString.m
function ids=IdFromString(string),
%IDFROMSTRING - output ids from a given string
%
%   Usage:
%      ids=IdFromString(string);
%
%   Examples:
%      ids=IdFromString('Parallel');
%      ids=IdFromString('79North');
%      ids=IdFromString('*');          %Print all tests

%Check input
if ~ischar(string)
	error('IdFromString error message: input argument is not a string');
end

%Initialize output
ids=[];

%Grep string
if strcmp(string,'*'),
	[dummy ids_raw]=system(['find ./ -name "test[0-9]*.m" | xargs grep "%Test Name:" | sed -e "s/test/ /g" -e "s/\.m:/ /g" | cut -d" " -f 2']);
else
	[dummy ids_raw]=system(['find ./ -name "test[0-9]*.m" | xargs grep "%Test Name:" | grep ' string ' | sed -e "s/test/ /g" -e "s/\.m:/ /g" | cut -d" " -f 2']);
end

%Check for error
if dummy~=0,
	error(['system call failed:' newline ids_raw]);
end

%return if no test found
if isempty(ids_raw),
	disp(['No test matches ''' string '''' ]);
	return
end

%Process string (delete return carriage);
ids_raw=strsplit_strict(ids_raw,char(10));
ids_raw=ids_raw(1:end-1);
ids=cellfun(@str2num,ids_raw);
ids=sort(ids);

%Display names
disp([ num2str(length(ids)) ' tests match ''' string '''']);
for i=1:length(ids)
	disp([ '   ' num2str(ids(i)) ' : ' IdToName(ids(i)) ]);
end

File: test/NightlyRun/IdFromString.py
#!/usr/bin/env python3

from GetAvailableTestIds import *
from IdToName import IdToName
import os
import re

# use verbose = False to print output when this is called by command line


def IdFromString(string, verbose=False):
    """IdFromString - output ids from a given string

    Usage:
        ids = IdFromString(string)
    
    Examples:
        ids = IdFromString('Dakota')
        ids = IdFromString('Slc')
        ids = IdFromString(' * ')
    """

    # Check input
    if not isinstance(string, str):
        raise TypeError('IdFromString error message: input argument is not a string.')
    string = string.replace("'", '')
    string = string.replace('"', '')

    # Get the test ids and names and scan for matches
    ids = []
    idnames = []
    available_test_ids = GetAvailableTestIds()
    for i in available_test_ids:
        name = IdToName(i)
        if (string == ' * ') or (name is not None and string in name):
            ids.append(i)
            idnames.append(name)

    # Return if no test found
    if not ids:
        print("No test matches '%s'." % string)
        return ids

    # Display names
    if verbose:
        idnames = [i for _, i in sorted(zip(ids, idnames), key=lambda pair: pair[0])]

    ids.sort()

    if verbose:
        print("{} tests match '{}':".format(len(ids), string))
        for i in range(len(ids)):
            print("   {} : {}".format(ids[i], idnames[i]))
    return ids
